numpy/lib/npyio.py

```diff
@@ -1071,6 +1071,15 @@ def read_data(chunk_size):
 
     dtype_types, packing = flatten_dtype_internal(dtype)
 
+    if usecols is not None:
+        if len(dtype_types) > 1 and len(usecols) != len(dtype_types):
+            raise ValueError('the number of columns specified in usecols '
+                             f'({len(usecols)}) does not equal the number of '
+                             f'fields in the given dtype ({len(dtype_types)})')
+        if len(usecols) == 0:
+            shp = (0, 0) if ndmin == 2 else (0,)
+            return np.empty(shp, dtype=dtype)
```

Comment:
I'd perhaps expect an a `(N, 0)` array (or maybe `(0, N)`?) where `N` is the number of lines in the file.

Reply:
I thought about doing that (specifically `(N, 0)`, or `(0, N)` if `unpack` is True).  But then I wondered if that was a foolish consistency.  However, if in fact most people would expect `(N, 0)`, then that's what it should do.

Reply:
Is this early exit even needed? What happens if you let the rest of the function run, to avoiding needing a special case?

Reply:
I think we should go with `(N, 0)`, ideally it works without any special handling. Speed is not relevant for such a weird corner case anyway.  It is a bit strange, but you could imaging reading `K` columns from multiple files and concatenating them or so, which only makes sense if the shape includes the `N` correctly.

Reply:
I'm working on an update to return (N, 0).  I just found a bug in how `loadtxt` handles nested structured dtypes, so I'll try to fix that, and then get back to this PR.

Reply:
For future reference, the bug that I encountered is https://github.com/numpy/numpy/issues/16678.